platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -298,7 +303,105 @@ namespace platoon_strategic_ihp
         return msg;
     }
 
-    // compose platoon info msg for all states
+    // ----------------------------- UCLA: helper functions for cut-in from front -------------------------------//
+
+    // Generate a maneuver ID for new maneuver plans (only used for compose lane change maneuver)
+    std::string PlatoonStrategicIHPPlugin::getNewManeuverId() const {
+        /**
+         * @brief Function to generate new maneuver ID. (Use for genreating new lane change maneuver)
+         * 
+         * @return A newly generated uuid as int.
+         */
+
+        static auto gen = boost::uuids::random_generator(); // Initialize uuid generator
+        
+        return boost::lexical_cast<std::string>(gen()); // generate uuid and convert to string
+    }
+
+    // Find target lane indexes based on downtracks for lane change (used for testing in a two-lane map).
```

Comment:
Is this method going to disappear completely?  If so, when?  If so, it should have a "TODO" tag on it to make sure we can easily find anything that needs to be cleaned up.